src/hotspot/cpu/x86/stubGenerator_x86_64_string.cpp

```diff
@@ -0,0 +1,2585 @@
+    __ bind(L_0x406019);
+
+    // CASE 0: CASE 1:
+    small_hs_jmp_table[0] = __ pc();
+    small_hs_jmp_table[1] = __ pc();
+    __ movq(rax, r15);
+    __ addptr(rsp, 0xf0);
+#ifdef _WIN64
+    __ pop(r9);
+    __ pop(r8);
+    __ pop(rcx);
+    __ pop(rdi);
+    __ pop(rsi);
+#endif
+    __ pop(rbp);
+    __ pop(rbx);
+    __ pop(r12);
+    __ pop(r13);
+    __ pop(r14);
+    __ pop(r15);
+    __ vzeroupper();
+
+    __ leave();  // required for proper stackwalking of RuntimeStub frame
+    __ ret(0);
+
+//  Tail cleanup stuff
+    __ bind(L_0x40602e);
+    __ movl(rax, rbp);
+    __ movq(r13, Address(rsp, 0x8));
+    __ subq(r13, Address(rsp, 0x18));
+    __ addq(r13, rax);
+    __ movq(rbx, Address(rsp, 0x20));
+    __ jmpb(L_exit);
+
+    __ bind(L_0x406044);
+    __ movq(rax, r15);
+    __ andq(rax, -32);
+    __ andl(r15, 0x1f);
+    __ movq(r13, -1);
+    __ cmpq(r15, rax);
+    __ jge(L_exit);
+    __ addq(rax, r11);
+
+    __ bind(L_0x40605e);
+    __ vpcmpeqb(xmm1, xmm0, Address(r11, r15, Address::times_1),
+                Assembler::AVX_256bit);
+    __ vpmovmskb(rcx, xmm1, Assembler::AVX_256bit);
+    __ testl(rcx, rcx);
+    __ jne_b(L_0x406093);
+    __ leaq(rcx, Address(r11, r15, Address::times_1));
+    __ addq(rcx, 0x20);
+    __ addq(r15, 0x20);
+    __ cmpq(rcx, rax);
+    __ jb(L_0x40605e);
+    __ jmp(L_exit);
+
+    __ bind(L_0x40607f);
+    __ tzcntl(rcx, rdx);
+    __ subq(rax, r11);
+    __ addq(rax, rcx);
+    __ movq(r13, rax);
+    __ jmp(L_exit);
+
+    __ bind(L_0x406093);
+    __ tzcntl(r13, rcx);
+    __ addq(r13, r15);
+    __ jmp(L_exit);
+
+    __ bind(L_0x4060a3);
+    __ movq(rbx, Address(rsp, 0x20));
+    __ movq(r13, -1);
+    __ jmp(L_exit);
+
+    ////////////////////////////////////////////////////////////////////////////////////////
+    ////////////////////////////////////////////////////////////////////////////////////////
+    ////////////////////////////////////////////////////////////////////////////////////////
+    __ align(8);
+
+    jump_table = __ pc();
+
+    for (jmp_ndx = 0; jmp_ndx < 32; jmp_ndx++) {
+      __ emit_address(large_hs_jmp_table[jmp_ndx]);
+    }
+
+    jump_table_1 = __ pc();
+
+    for (jmp_ndx = 0; jmp_ndx < 32; jmp_ndx++) {
+      __ emit_address(small_hs_jmp_table[jmp_ndx]);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////
+    ////////////////////////////////////////////////////////////////////////////////////////
+    ////////////////////////////////////////////////////////////////////////////////////////
+
+    __ align(16);
+    __ bind(L_begin);
+    __ push(r15);
+    __ push(r14);
+    __ push(r13);
+    __ push(r12);
+    __ push(rbx);
+    __ push(rbp);
+#ifdef _WIN64
+    __ push(rsi);
+    __ push(rdi);
+    __ push(rcx);
+    __ push(r8);
+    __ push(r9);
+
+    __ movq(rdi, rcx);
+    __ movq(rsi, rdx);
+    __ movq(rdx, r8);
+    __ movq(rcx, r9);
+#endif
```

Comment:
Can we spill them into XXMs, to save costly stack operations.

Reply:
Changed.